tests/iostability/test_iostability_degraded_path.py

```diff
@@ -0,0 +1,189 @@
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+"""Test Suite for IO stability Happy Path workloads."""
+import logging
+import os
+import random
+import time
+from datetime import datetime, timedelta
+
+import pytest
+
+from commons import configmanager
+from commons.constants import K8S_SCRIPTS_PATH, POD_NAME_PREFIX
+from commons.helpers.health_helper import Health
+from commons.helpers.pods_helper import LogicalNode
+from commons.params import LATEST_LOG_FOLDER
+from commons.utils import assert_utils, system_utils, support_bundle_utils
+from config import CMN_CFG
+from config.s3 import S3_CFG
+from conftest import LOG_DIR
+from libs.ha.ha_common_libs_k8s import HAK8s
+from libs.s3 import ACCESS_KEY, SECRET_KEY
+from libs.s3.s3_test_lib import S3TestLib
+from scripts.s3_bench import s3bench
+
+
+class TestIOWorkloadDegradedPath:
+    """Test suite for IO Stability in Degraded path."""
+
+    @classmethod
+    def setup_class(cls):
+        """Setup class."""
+        cls.log = logging.getLogger(__name__)
+        cls.num_nodes = len(CMN_CFG["nodes"])
+        cls.master_node_list = []
+        cls.worker_node_list = []
+        cls.hlth_master_list = []
+        for node in CMN_CFG['nodes']:
+            node_obj = LogicalNode(hostname=node["hostname"],
+                                   username=node["username"],
+                                   password=node["password"])
+            if node["node_type"].lower() == "master":
+                cls.master_node_list.append(node_obj)
+                cls.hlth_master_list.append(Health(hostname=node["hostname"],
+                                                   username=node["username"],
+                                                   password=node["password"]))
+            else:
+                cls.worker_node_list.append(node_obj)
+        cls.ha_obj = HAK8s()
+        cls.test_cfg = configmanager.get_config_wrapper(fpath="config/iostability_test.yaml")
+        cls.setup_type = CMN_CFG["setup_type"]
+        cls.s3t_obj = S3TestLib(access_key=ACCESS_KEY, secret_key=SECRET_KEY)
+        cls.test_completed = False
+
+    def teardown_class(self):
+        """Teardown class"""
+        if not self.test_completed:
+            self.log.info("Test Failure observed, collecting support bundle")
+            path = os.path.join(LOG_DIR, LATEST_LOG_FOLDER)
+            resp = support_bundle_utils.collect_support_bundle_k8s(local_dir_path=path,
+                                                                   scripts_path=K8S_SCRIPTS_PATH)
+            assert_utils.assert_true(resp)
+
+    def execute_workload_distribution(self, distribution, clients, total_obj,
```

Comment:
Is it a good candidate for moving to lib as I guess it will be used by all io stability files?